saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/authentication/OpenSamlAuthenticationRequestResolver.java

```diff
@@ -119,6 +117,11 @@ <T extends AbstractSaml2AuthenticationRequest> T resolve(HttpServletRequest requ
 		authnRequest.setIssuer(iss);
 		authnRequest.setDestination(registration.getAssertingPartyDetails().getSingleSignOnServiceLocation());
 		authnRequest.setAssertionConsumerServiceURL(registration.getAssertionConsumerServiceLocation());
+		if (registration.getNameIdFormat() != null) {
+			NameIDPolicy nameIdPolicy = new NameIDPolicyBuilder().buildObject();
```

Comment:
Instead of constructing a builder on the fly, please retrieve it from the `XMLObjectProviderRegistry` and save it as a class member variable. You can see other builders in the constructor as examples.